src/RequestConditionalPlugin.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Http\Message\RequestMatcher;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Apply a delegated plugin based on a request match.
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+final class RequestConditionalPlugin implements Plugin
```

Comment:
Alternative could be RequestMatcherPlugin. It's kinda obvious, don't know why I didn't think about it earlier.